core/app/login/LoginForm.tsx

```diff
@@ -34,17 +36,21 @@ export default function LoginForm() {
 	const runLoginWithPassword = useServerAction(actions.loginWithPassword);
 
 	const handleSubmit = async (formData: z.infer<typeof loginFormSchema>) => {
-		await runLoginWithPassword({
+		const result = await runLoginWithPassword({
 			email: formData.email,
 			password: formData.password,
 			redirectTo: searchParams.get("redirectTo") ?? null,
 		});
+
+		if (result && result.error) {
+			form.setError("root", { message: result.error });
+		}
```

Comment:
im not super happy about this, but this is basically a way of signaling to `FormSubmitButton` that there's a diff between "the submit action ran without error, but the sign in was unsuccessful" and "the user signed in successfully"